exchange/index.go

```diff
@@ -453,6 +469,97 @@ func (idx *Index) evict(size uint64) uint64 {
 	return evicted
 }
 
+// tagForGC tags CIDs that will be evicted during garbage collection
+func (idx *Index) tagForGC(ref *DataRef) error {
+	idx.emu.Lock()
+	defer idx.emu.Unlock()
+
+	return utils.WalkDAG(context.TODO(), ref.PayloadCID, idx.bstore, sel.All(), func(block blocks.Block) error {
+		idx.gcSet.Add(block.Cid())
+		return nil
+	})
+}
+
+// GC removes tagged CIDs
+func (idx *Index) GC() error {
+	idx.emu.Lock()
+	defer idx.emu.Unlock()
+
+	// exit if there is nothing to evict
+	if idx.gcSet.Len() == 0 {
+		return nil
+	}
+
+	gcds, ok := idx.ds.(datastore.GCDatastore)
+	if !ok {
+		return errors.New("datastore is not a GCDatastore")
+	}
+	err := gcds.CollectGarbage()
```

Comment:
You wanna call that **after** deleting the blocks

Reply:
Indeed my bad, I forgot that blockstore encapsulates a datastore